packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -68,56 +71,76 @@ export function populateOutputBundleOptions(outputPaths: OutputPaths): OutputPat
       stdio: ["inherit", "pipe", "pipe"],
     });
-    let outputPathOptions = {} as OutputPathOptions;
+    let buildOutput = "";
     let manifest = {} as ValidManifest;
 
-    if (childProcess.stdout) {
-      childProcess.stdout.on("data", (data) => {
-        try {
-          if (data.toString().includes("outputPaths")) {
-            const parsedManifest = JSON.parse(data);
-            // validate if the manifest is of the expected form
-            manifest = buildManifestSchema.parse(parsedManifest);
-            if (manifest["errors"].length > 0) {
-              // errors when extracting manifest
-              manifest.errors.forEach((error) => {
-                logger.error(error);
-              });
-            }
-            if (manifest["warnings"].length > 0) {
-              // warnings when extracting manifest
-              manifest.warnings.forEach((warning) => {
-                logger.info(warning);
-              });
-            }
-            outputPathOptions = populateOutputBundleOptions(manifest["outputPaths"]);
-          }
-        } catch (error) {
-          throw new Error("Build manifest is not of expected structure: " + error);
-        }
-      });
-    } else {
-      throw new Error("Unable to locate build manifest with output paths.");
-    }
+    childProcess.stdout.on("data", (data: Buffer) => {
+      buildOutput += data.toString();
+    });
 
     childProcess.on("exit", (code) => {
-      if (code === 0) return resolve(outputPathOptions);
-      reject();
+      if (!buildOutput) {
+        reject(new Error("Unable to locate build manifest with output paths."));
+      }
+      if (code !== 0) {
+        reject(new Error(`Process exited with error code ${code}`));
+      }
+      try {
+        const strippedManifest = extractManifestOutput(buildOutput);
+        const parsedManifest = JSON.parse(strippedManifest) as string;
+        // validate if the manifest is of the expected form
+        manifest = buildManifestSchema.parse(parsedManifest);
+        if (manifest["errors"].length > 0) {
+          // errors when extracting manifest
+          manifest.errors.forEach((error) => {
+            logger.error(error);
+          });
+        }
+        if (manifest["warnings"].length > 0) {
+          // warnings when extracting manifest
+          manifest.warnings.forEach((warning) => {
+            logger.info(warning);
+          });
+        }
+        resolve(populateOutputBundleOptions(manifest["outputPaths"]));
+      } catch (error) {
+        reject(new Error("Build manifest is not of expected structure: " + error));
+      }
     });
   });
 
-/* 
-Move the base output directory, which contains the server and browser bundle directory, and prerendered routes
-as well as generating bundle.yaml.
+/**
+ * Extracts the build manifest from the build command's console output.
+ * N.B. Unfortunately, there is currently no consistent way to suppress extraneous default output from the task
```

Comment:
ooc what is nx sending to std out?

Reply:
Looks like something like this:
<img width="797" alt="Screenshot 2024-03-22 at 4 54 21 PM" src="https://github.com/FirebaseExtended/firebase-framework-tools/assets/112491344/3e0ab850-d906-4208-999a-3a88cd8370b2">